Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -871,6 +871,55 @@ void ParseMarvinSmartsLine(RWMol *mol, const std::string &text,
   }
 }
 
+void ParseAttachPointLine(RWMol *mol, const std::string &text, unsigned int line) {
+  PRECONDITION(mol, "bad mol");
+  PRECONDITION(text.substr(0, 6) == std::string("M  APO"), "bad APO line");
+
+  unsigned int nent;
+  try {
+    nent = FileParserUtils::stripSpacesAndCast<unsigned int>(text.substr(6, 3));
+  } catch (boost::bad_lexical_cast &) {
+    std::ostringstream errout;
+    errout << "Cannot convert '" << text.substr(6, 3) << "' to int on line "
+           << line;
+    throw FileParseException(errout.str());
+  }
+  unsigned int spos = 9;
+  for (unsigned int ie = 0; ie < nent; ie++) {
+    unsigned int aid = 0;
+    int val = 0;
+    try {
+      aid = FileParserUtils::stripSpacesAndCast<unsigned int>(
+          text.substr(spos, 4));
+      spos += 4;
+      if (text.size() >= spos + 4 && text.substr(spos, 4) != "    ") {
+        val = FileParserUtils::stripSpacesAndCast<int>(text.substr(spos, 4));
+      }
```

Comment:
I'm not checking these for correctness since we aren't actually doing anything with them.
I suppose it doesn't hurt to actually check though. :-)

I will also directly translate "-1" -> "3".

Reply:
Sorry, maybe I didn't explain myself correctly. It is actually the opposite: The atom block uses "-1", the "M APO" line uses 3. So, it would actually be the opposite: "3" -> "-1".

Reply:
no that's my fault, I managed to confuse myself about which section of the docs I was reading.